### PR TITLE
Unbracketed `cond`

### PR DESCRIPTION
**Pull request by Kodiologist**
Partly fulfills #2210.

### CODE REVIEW

File: tests/native_tests/language.hy [hunk 1]
```diff
@@ -340,19 +340,29 @@
...
+    (is None None) (do (setv x True) (assert x)))
   (assert (= (cond) None))
 
   (assert (= (cond
-    [False]
-    [[]]
-    [8])) 8)
+    False 1
+    [] 2
+    True 8)) 8)
```
Comment (allison-casey) [inline tests/native_tests/language.hy:350]: so it turns out this paren is one short and we need to slurp that 8 up to actually check equality

File: tests/native_tests/language.hy [hunk 2]
```diff
@@ -340,19 +340,29 @@
 
 (defn test-cond []
   (cond
-   [(= 1 2) (assert (is True False))]
-   [(is None None) (setv x True) (assert x)])
+    (= 1 2) (assert (is True False))
+    (is None None) (do (setv x True) (assert x)))
   (assert (= (cond) None))
```
Comment (allison-casey) [inline tests/native_tests/language.hy:345]: also can we get this `=` changed to `is` while we're here?

**Review by allison-casey — COMMENTED** [review]
Fantastic! this is pretty much what I had in my branch. Everything looks good! just noticed two things we might as well fix in the cond test unrelated to this PR we might as well pick up while we're here. Also, I have a branch with the strict if form that would finish the conditional forms refactor we've been talking about if you want to yoink that and finish it up while i work on those sphinx hydomain issues you brought up

**Comment by Kodiologist** [comment]
> this is pretty much what I had in my branch.

Oops, I did this because I thought it was the part you hadn't already done.

> Also, I have a branch with the strict if form that would finish the conditional forms refactor we've been talking about if you want to yoink that

Okay. Let's do it as another PR.

File: tests/native_tests/language.hy (hunk 1 above)
Comment (Kodiologist) [reply tests/native_tests/language.hy]: Good catch. Fixed.

File: tests/native_tests/language.hy (hunk 2 above)
Comment (Kodiologist) [reply tests/native_tests/language.hy:345]: You mean changing `(= 1 2)` to `(is 1 2)`? Aren't the semantics of `is` with numbers a bit dicey?

Comment (allison-casey) [reply tests/native_tests/language.hy:345]: Oh no the `(= (cond) None)` one

Comment (Kodiologist) [reply tests/native_tests/language.hy]: Oh, right. My eyeballs just skipped over that line because I hadn't edited it. Done.

**Review by allison-casey — APPROVED** [review]

**Pull request merged**